src/compton/Compton.cc

```diff
@@ -33,16 +33,20 @@ namespace rtt_compton {
  *
  * \param[in] filehandle The name of the Compton multigroup file
  */
-Compton::Compton(const std::string &filehandle) {
+Compton::Compton(const std::string &filehandle, const bool llnl_style) {
 
   // Make a compton file object to read the multigroup data
   compton_file Cfile(false);
-
-  // initialize the electron temperature interpolator with the mg compton data
-  ei.reset(new etemp_interp(Cfile.read_mg_csk_data(filehandle)));
-
-  // Make sure the SP exists...
-  Ensure(ei);
+  if (!llnl_style) {
+    // initialize the electron temperature interpolator with the mg compton data
+    ei.reset(new etemp_interp(Cfile.read_mg_csk_data(filehandle)));
+    // Make sure the SP exists...
```

Comment:
Please review the documentation for smart pointer to see if this is the `new` form that you want here.  It might be better to use `make_shared`, `make_unique`, or some other form.

Reply:
Yeah, I think I can do away with the shared pointers altogether -- I will retool this. Thanks!